sourcegraph/sourcegraph
# Wildcard: Fix disabled target tooltip (fix disabled button state)

Fixes https://github.com/sourcegraph/sourcegraph/issues/41683
Fixes https://github.com/sourcegraph/sourcegraph/issues/38782

## Background 
Prior to this PR Button rendered `<button />` element with disable attribute. This attribute (disabled) gives the ability to turn on/off button clicks and apply special disabled button visual styles. However, the `disabled=true` attribute turns off all other events besides the click event. This adds a lot of problems to all other components that somehow want to track user events on the disabled button. For example, the tooltip prior to this PR didn't work without special disable state handling. 

In this PR, we remove the disabled tooltip and instead of this, we add `aria-disabled=true`. Plus to this, we add a custom onClick handler in order to manually turn on/off click events. 

## Test plan
- Make sure that there is no way to click the disabled button 
- Make sure that the tooltip works properly even with the disabled button. 



## App preview:

- [Web](https://sg-web-vk-fix-disabled-target-tooltip.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
